### PR TITLE
Copying ICSharpCode.AvalonEdit.dll to Dynamo 2.0 bin folders disabled

### LINKED ISSUE #291: ICSharpCode.AvalonEdit.dll gets copied over to Dynamo 2.0 bin folder
#### Description:

As in the title: the file causing the weird behaviour reported in #289 is still present in _%AppData%\Dynamo\Dynamo Revit\2.0\packages\BHoM\bin_ both after compilation as well as after installation from today's Alpha. My intuitive bet is that it is a Dynamo_Toolkit issue, but please feel free to transfer to the installer if needed.

Dynamo 2.3 works fine thanks to different assembly loading mechanism.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #291

### Test files

Not needed - compile and check whether _ICSharpCode.AvalonEdit.dll_ gets copied over to Dynamo 2.0 bin folders.

### Changelog

- copying ICSharpCode.AvalonEdit.dll to Dynamo 2.0 bin folders disabled

### CODE REVIEW

**Comment by pawelbaran** [comment]
@FraserGreenroyd I expect this PR being unhappy about the _excludes.txt_ file - happy for you to step in and reshuffle the files so that the CI is happy if needed

File: Dynamo20/Dynamo_UI/Dynamo20_UI.csproj
Comment (FraserGreenroyd) [inline Dynamo20/Dynamo_UI/Dynamo20_UI.csproj:278]:
```suggestion
xcopy /Y /E /I C:\ProgramData\BHoM\Assemblies\*.dll "C:\Users\$(Username)\AppData\Roaming\Dynamo\Dynamo Core\2.0\packages\BHoM\bin" /EXCLUDE:"$(ProjectDir)excludes.txt"
```

File: Dynamo20/Dynamo_UI/Dynamo20_UI.csproj
Comment (FraserGreenroyd) [inline Dynamo20/Dynamo_UI/Dynamo20_UI.csproj:284]:
```suggestion
xcopy /Y /E /I C:\ProgramData\BHoM\Assemblies\*.dll "C:\Users\$(Username)\AppData\Roaming\Dynamo\Dynamo Revit\2.0\packages\BHoM\bin" /EXCLUDE:"$(ProjectDir)excludes.txt"
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Needs speech marks for machines that might have spaces in the file path.

But this doesn't work for me, when trying to build the repo on top of todays alpha, it fails to read the `excludes.txt` file, and doesn't put any `bin` folder at all within the Dynamo20 structure.

File: Dynamo20/Dynamo_UI/Dynamo20_UI.csproj
Comment (pawelbaran) [reply Dynamo20/Dynamo_UI/Dynamo20_UI.csproj:278]: Yeah it fails to read excludes.txt if the path is wrapped in "" - that was my original idea but was failing as described by you. Not sure what to do then?

**Comment by pawelbaran** [comment]
@FraserGreenroyd I have just pushed a solution suggested by @adecler, namely removal of AvalonEdit after xcopy instead of excuding it from copying. Should be ready for re-review now.

**Comment by pawelbaran** [comment]
@BHoMBot check compliance

File: Dynamo20/Dynamo_UI/Dynamo20_UI.csproj
Comment (adecler) [inline Dynamo20/Dynamo_UI/Dynamo20_UI.csproj:279]:
```suggestion
del "C:\Users\$(Username)\AppData\Roaming\Dynamo\Dynamo Core\2.0\packages\BHoM\bin\ICSharpCode.AvalonEdit.dll"
```

**Review by FraserGreenroyd — APPROVED** [review]
I ran the PR as stated, and the file `ICSharpCode.AvalonEdit.dll` is removed from the 2.0 Dynamo folders.

Whether this will work for the installer will remain to be seen but I think it will because of how it scrapes things for the Dynamo folders.

I do have a `ICSharpCode.Decompiler.dll` in the folder, suspect that ones coming from a BHoM_UI dependency possibly and not sure if this is causing any issues with Dynamo. It may not have come from BHoM and could be a hang over of something else. Just flagging in case.

I have not opened Dynamo or tried to run a script. I have only followed what was on the PR list and confirmed the removal of the DLL visually.

Would like @adecler to have final review before merge though as it's UI in case I've missed something.

**Review by adecler — APPROVED** [review]
I am happy with this. Works as intended.

@FraserGreenroyd , I don't have an issue with `ICSharpCode.Decompiler.dll` being there as Dynamo doesn't mind about that one. The issue we had was with dynamo using another version of AvalonEdit  for the Code Block component.

**Comment by adecler** [comment]
@BHoMBot check copyright

**Comment by adecler** [comment]
@BHoMBot check installer

**Comment by adecler** [comment]
@BHoMBot check installer

**Pull request merged**
